### PR TITLE
Avoid excess dependencies in `syntax/parse/lib/function-header`.

### PR DESCRIPTION
**Pull request by samth**
This causes `racket/match` to no longer depend on `racket/contract`.

cc @rmculpepper

### CODE REVIEW

**Comment by shhyou** [comment]
The `??` after `match*/derived` needs to be `~?`.

File: racket/collects/racket/match/define-forms.rkt [hunk 1]
```diff
@@ -197,7 +196,7 @@
      (define-syntax (define/match stx)
        (syntax-parse stx
          [(_ ?header:function-header ?clause ...)
-          (quasitemplate
+          (quasisyntax
            (define ?header
              (match*/derived (?? ?header.params) #,stx
```
Comment (samth) [inline racket/collects/racket/match/define-forms.rkt:201]:
```suggestion
             (match*/derived (~? ?header.params) #,stx
```

**Comment by samth** [comment]
@rmculpepper is this ok to merge?

**Comment by rmculpepper** [comment]
LGTM

**Pull request merged**